### PR TITLE
add communities, roles, check permissions, improve tests

### PR DESCRIPTION
**Pull request by emanueldima**

### CODE REVIEW

File: gefserver/server/permissions.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	Response{a.w}.Forbidden("A service can only be edited by its owner")
+	return
+}
+
+func (a Authorization) allowCreateJob() (allow bool, user *db.User) {
+	allow, user = a.getUserInfo()
+	if user == nil || allow {
+		return
+	}
+	// roles, err := a.s.db.GetUserRoles(user.ID)
```
Comment (ubuntolog) [inline gefserver/server/permissions.go]: Unused code

File: gefserver/main.go [hunk 2]
```diff
@@ -10,6 +10,13 @@ import (
 	"github.com/EUDAT-GEF/GEF/gefserver/server"
 )
 
+// TODO UI:
```
Comment (emanueldima) [inline gefserver/main.go]: You're right, I forgot this

File: gefserver/server/access.go [hunk 3]
```diff
@@ -131,29 +120,75 @@ func (s *Server) removeTokenHandler(w http.ResponseWriter, r *http.Request) {
 	Response{w}.Ok(jmap("Token", token))
 }
 
-func (s *Server) getLoggedInUser(r *http.Request) (*db.User, error) {
+///////////////////////////////////////////////////////////////////////////////
```
Comment (emanueldima) [inline gefserver/server/access.go]: I'm using these lines to separate code sections, this makes it more clear to me, but you can remove them if you don't like them 😄

File: gefserver/server/permissions.go (hunk 1 above)
Comment (emanueldima) [inline gefserver/server/permissions.go]: This one should stay for the moment, it's how the function should ideally work.

**Pull request merged**